File: frontend/lib/data/models/property.dart
import 'mark_type.dart';

class Property {
  String? listingURL;
  dynamic? size;
  double? ocrSize;
  double? longitude;
  double? latitude;
  String? imageURL;
  String? status;
  String? propertyType;
  dynamic? price;
  String? displayableAddress;
  List<dynamic>? floorPlan;
  MarkType? markType;

  Property(this.listingURL, this.size, this.imageURL, this.price,
      this.displayableAddress);

  Property.fromJson(Map<String, dynamic> json)
      : listingURL = json['details_url'] ?? 'No listing URL',
        ocrSize = json['ocr_size'],
        status = json['status'] ?? 'No status',
        longitude = json['longitude'] ?? 'No longitude',
        latitude = json['latitude'] ?? 'No latitude',
        propertyType = json['property_type'] ?? 'No property type',
        imageURL = json['image_url'] ?? 'No image',
        price = json['price'] ?? 'No price',
        displayableAddress = json['displayable_address'] ?? 'No Address',
        floorPlan = json['floor_plan'] ?? [],
  markType = markTypeFromJSON(json['mark']);
}

File: backend/app/schemas/property_.py
from typing import List, Optional

from pydantic import BaseModel, Field

from app.models.property_ import SaveMark
from app.schemas.station import Station


class PostcodeList(BaseModel):
    postcodes: List[str]


class Property(BaseModel):
    listing_url: str = Field(alias="details_url")
    ocr_size: Optional[float] = None
    longitude: float
    latitude: float
    image_url: Optional[str] = None
    status: Optional[str] = None
    property_type: Optional[str] = None
    price: Optional[int] = None
    displayable_address: Optional[str] = None
    floor_plan: Optional[List[str]] = None
    # stations: Optional[List[Station]] = None
    mark: Optional[SaveMark] = None

    def __lt__(self, other: "Property"):
        return self.ocr_size < other.ocr_size


class PropertyList(BaseModel):
    properties: List[Property]
